### ISSUE TITLE
Sidebar behaviour is problematic when you prefer it to be collapsed

### ISSUE THREAD

**Issue by uglycoyote** [buildbot/guanlecoja-ui#9]
When viewing buildbot pages which are dense with information, it is helpful to collapse the side-bar.  

The current implementation is a bit unfriendly to users who wish to have it always collapsed, in two regards:

1.  Upon reloading the page, it opens again
2. When pinned, it expands and contracts as your mouse passes over it,  which is problematic because it is a vertical bar and easy to mouse-over accidentally.  e.g., if buildbot's browser window is on the right monitor in a multi-monitor setup.

I have a local commit which mostly fixes these problems, but I assume it would not be accepted as a pull request because i have hard-coded the side bar to always default to "unpinned/collapsed", which might be everyone's preference.  

Ideally, upon pinning or unpinning, it should set a setting which persists across sessions.  This way the next time you open the page it will remember the pinned vs collapsed state that you used previously.

If you can advise on how to make a persistent setting, perhaps that can be added.  I have added custom settings to my buildbot plugins but as far as I can tell the guanlecoja-ui does not have direct support for this.  

https://github.com/bb-nl/guanlecoja-ui/commit/24370a40c6bcd37dc2f85268ba5f6a82a0154808

**Comment by uglycoyote** [comment]
to add to the above, I made an attempt at getting this to be more settings driven in this commit
https://github.com/bb-nl/guanlecoja-ui/commit/314ae6acd160d1d828befebf2fece1b38d5f34db

Then from `app.route.coffee` in the buildbot repository, the following code works:

```
glMenuServiceProvider.setPinned(false)  # should use a setting here!!!  but can't add bbSettingsService to constructor!
glMenuServiceProvider.setPinnedChangedCallback( (value)=>console.log("pinned changed to", value) )
```

I attempted to hook this up to an actual setting, but adding `bbSettingsService` as a constructor argument to `class Route` did not work (unknown provider).  I don't understand Angular well enough to know why `bbSettingsService` can be used as a dependency of various controllers like `class Builders` but not here.  Using ``bbSettingsServiceProvider`` in the constructor works, but that seems to be a different object which only gives me access to the definitions of the settings and not the current values of them.

**Comment by tardyp** [comment]
guanlecoja-ui is independant from build, so that's why it is not usable there.

You could use an independent localStorage key in order to store that preference.
https://developer.mozilla.org/fr/docs/Web/API/Window/localStorage

**Comment by uglycoyote** [comment]
Yes, I realized already that guanlecoja-ui was independent from buildbot -- my plan was to expose 1) a way to set the initial collapsed/expanded state 2) a way for buildbot to get an onChanged callback so it could write the new preference into the settings.  The [app.route.coffee](https://github.com/buildbot/buildbot/blob/master/www/base/src/app/app.route.coffee) that I was mentioning is part of the buildbot sources, but for reason bbSettingsService was still not accessible there when I tried to add it to the constructor.

Local Storage sounds like a good alternative though!
